Get the SMS rate from the API

quis opened the pull request:
Copies the approach we used for letters from
- https://github.com/alphagov/notifications-admin/pull/4960

Gets the rates from:
- https://github.com/alphagov/notifications-api/pull/4049

Caches them in Redis for 1 hour so we don’t hit the API on high-traffic public pages (for example the homepage)

Uses the current rate as test data so the existing tests pass unmodified.

Adds a conditional to only show incoming pricing message when old pricing in effect.

CrystalPea commented on app/templates/views/guidance/pricing/text-message-pricing.html:
```diff
@@ -17,16 +17,21 @@ <h1 class="heading-large">Text message pricing</h1>
   <p class="govuk-body">When a service has used its annual allowance, it costs {{ sms_rate }} pence (plus VAT) for each text message you send.</p>
 
-  <div class="govuk-inset-text">  
+  {% if sms_rate != 2.27 %}
```
We will need to remember to delete that all if statement next week, right?

quis replied:
Yup, I will put up a PR to delete it once this is merged

quis commented:
Thanks @CrystalPea. I have:
- fixed the pounds/pence formatting in 1339fa214c8a5b7dde61e73bf1aaa92d20a2d612
- added the SMS rates to the cache-clearing form in b0ca67b19c87a838b63cb568d734322ef9bc87ea

CrystalPea reviewed (approved):
Looks good, ship it! I tested cache clearing, too 😁 

![Screenshot 2024-03-27 at 15 25 03](https://github.com/alphagov/notifications-admin/assets/20957548/76ec3900-4202-4938-a97c-991ce767d5f5)

karlchillmaid reviewed (approved):
Content changes look correct.

I like the addition of logic to display the right content based on the current SMS rate.

The tests look consistent with the code changes.
